app/controllers/criteria_controller.rb

```diff
@@ -36,6 +36,7 @@ def create
   end
 
   def edit
+    @assignment = Assignment.find(params[:assignment_id])
```

Comment:
This is not necessary. You can access `@criterion.assignment` through the assignment. In general you should aim to have few instance variables.